pkg/controllers/provisioning/scheduling/volumetopology.go

```diff
@@ -181,15 +182,32 @@ func (v *VolumeTopology) ValidatePersistentVolumeClaims(ctx context.Context, pod
 				continue
 			}
 
-			storageClassName = pvc.Spec.StorageClassName
+			storageClassName = lo.FromPtr(pvc.Spec.StorageClassName)
 			volumeName = pvc.Spec.VolumeName
 		} else if volume.Ephemeral != nil {
```

Comment:
Is it possible to just rely on the PVC directly always? The volumeClaimTemplate generates a PVC which has a known name. We should be able to look-up that known name and then use that to check the storage class name and the volume name. I believe in that case, the storage class name should never not be set and we can handle the cases which are:
1.  The storage class name is empty and the volume name needs to exist
2. The storage class name is not empty but the volume name exists (so this overrides the storage class in this case)
3. The storage class name is not empty AND the volume name is empty (so we should check if the storage class exists)

Reply:
We should be able to, thats what I was calling out in the TODO that I was planning to do in a follow-up PR. It does make sense to bring in scope here though,  I don't think I did because I tied it to removing default storage class discovery.